### PR TITLE
Add ASL validation

### PR DESCRIPTION
**Pull request by zklamm**
I've added the npm module `asl-validator` to the project.

Now, whenever the `generateStateMachineParams` function is invoked and the state machine definition is invalid, the proper errors will be logged to the console.

### CODE REVIEW

File: src/aws/step-function/generateStateMachineParams.js [hunk 1]
```diff
@@ -1,29 +1,25 @@
...
-  };
+  if (isValid) {
+    return {
+      definition,
+      name: workflowName,
+      roleArn: role.Role.Arn,
+    };
+  } else {
+    console.error("✕ State machine definition is invalid:", errorsText("\n"));
+    return;
```
Comment (johnisom) [inline src/aws/step-function/generateStateMachineParams.js:21]: This is good, but the SDK still tries to create the state machine, just with an empty definition instead.

Comment (johnisom) [reply src/aws/step-function/generateStateMachineParams.js:21]: There should probably be some validation in the `src/aws/step-function/createStepFunction.js` file. Maybe this logic should be moved there?

Comment (zklamm) [reply src/aws/step-function/generateStateMachineParams.js]: Good point. I've moved the logic to `createStepFunctions.js` instead.

File: src/aws/step-function/createStepFunction.js [hunk 2]
```diff
@@ -1,7 +1,16 @@
...
+
+  if (isValid) {
+    return stepFunctions.createStateMachine(params).promise();
+  } else {
+    console.error("✕ State machine definition is invalid:", errorsText("\n"));
+    throw new Error();
+  }
 };
 
 module.exports = createStepFunction;
```
Comment (johnisom) [inline src/aws/step-function/createStepFunction.js:16]: :+1:

**Pull request merged**